### PR TITLE
Fixes for hound on windows

### PR DESCRIPTION
**Pull request by ameerkat**
Makes hound build and run on windows. I noticed there was an existing PR from February where there was no more progress being made so I'm suggesting the changes I made to make it work on Windows.

### CODE REVIEW

**Comment by kellegous** [comment]
Thanks. I've been traveling a bit which has kept me from reviewing this. I should be able to take a look this weekend or early next week.

**Comment by KindDragon** [comment]
:+1: 
We can I find Hound build for Windows?

**Comment by ameerkat** [comment]
I added a windows build here if you want to try it out https://github.com/ameerkat/Hound/releases/tag/v0.2.1-win built/tested on windows 8.1

**Comment by KindDragon** [comment]
`file://` protocol doesn't work
```
2015/05/27 13:38:19 exit status 128                                                 
2015/05/27 13:38:19 Searcher started for trunk                                      
2015/05/27 13:38:19 Failed to clone file:///c:/Work/TDCode/trunk/, see output below 
Cloning into 'vcs-502bf35e806e9ba2b99c1c0c6f64ea25bb54bd6c'...                      
fatal: 'c:/Work/TDCode/trunk/' does not appear to be a git repository               
fatal: Could not read from remote repository.                                       
                                                                                    
Please make sure you have the correct access rights                                 
and the repository exists.                                                          
Continuing...                                                                       
```

**Comment by kellegous** [comment]
Took me a little while to get a windows Vm setup.

index tests are currently failing with the following output:

```
2015/05/27 07:55:52 21858 data bytes, 50995 index bytes
2015/05/27 07:55:52 UnmapViewOfFile C:\Users\IEUser\AppData\Local\Temp\hound6118
65747\tri: Attempt to access invalid address.
FAIL    github.com/etsy/Hound/index     0.117s
```

Haven't looked into it yet to know if it's a problem w/ the test or if there is an issue unmapping.

Should be reproducible with:
`go test github.com/etsy/hound/...`

**Comment by ameerkat** [comment]
@KindDragon Are you sure you are pointing to a directory with a .git folder in it? It can not be a subfolder. File protocol works fine for me in that case.

@kellegous Will take a look, I haven't seen that come up during normal operation.

**Comment by ameerkat** [comment]
@kellegous Didn't get a chance to get some time to fix this until now. All the tests are now passing on my machine. There were two main problems, an omission of [0] in the first pointer value in un memory mapping the file in windows and during TestRemove the product code did not close the main index "tri" file resulting in the file being locked when trying to delete the directory. After going through the code I found no place that it was closed and noticed indexAllFiles doesn't return a reference to it so I closed it there which resolved the 2nd issue. This does change the functionality on other platforms so I would recommend running the tests on them as I do not have easy access to a box to set this up.

File: codesearch/index/mmap_windows.go [hunk 1]
```diff
@@ -33,5 +33,25 @@ func mmapFile(f *os.File) mmapData {
...
 	}
 	data := (*[1 << 30]byte)(unsafe.Pointer(addr))
-	return mmapData{f, data[:size]}
+	return mmapData{f, data[:size], data[:]}
 }
+
+func unmmapFile(m *mmapData) error {
+	err := syscall.UnmapViewOfFile(uintptr(unsafe.Pointer(&m.d[0])))
+	if err != nil {
+		log.Fatalf(err)
```
Comment (kellegous) [inline codesearch/index/mmap_windows.go]: Hm, do we want to os.Exit(1) here? `log.Fatal` is truly a fatal and aborts the process.

**Comment by kellegous** [comment]
This looks good, thanks for doing this. I'm going to run this on my VM soon to try it out.

File: codesearch/index/mmap_windows.go (hunk 1 above)
Comment (ameerkat) [inline codesearch/index/mmap_windows.go]: @kellegous Ah true, we should at least keep consistency with the other unmmap implemenetations. I'll change it

**Comment by kellegous** [comment]
This is still crashing for me when a repository is rebuilt (when a new commit arrives).

```
2015/06/09 12:30:08 Searcher started for cockroachdb
2015/06/09 12:30:16 merge 0 files + mem
2015/06/09 12:30:16 7825085 data bytes, 1581191 index bytes
2015/06/09 12:30:16 All indexes built!
2015/06/09 12:30:16 running server at http://localhost:6080...
2015/06/09 12:32:20 Rebuilding cockroachdb for fa9607b60aec1cacc1ef56640bd383f97
41254f5
2015/06/09 12:32:24 merge 0 files + mem
2015/06/09 12:32:24 7825093 data bytes, 1581208 index bytes
2015/06/09 12:32:24 UnmapViewOfFile C:\Users\IEUser\hound\data\idx-e5f688621c184
7f6\tri: Attempt to access invalid address.
```

**Comment by ameerkat** [comment]
Ah, don't think I tried out that scenario. I'll take a look sometime later this week. Would be good to add a test that would catch this.

**Comment by ameerkat** [comment]
mmap in windows was creating a handle and it never got closed which ended up locking that file indefinitely. I closed it in mmap since it isn't passed out anywhere. Looks like it's working as I'm getting the following when pushing changes to the repo

2015/06/09 23:28:02 Searcher started for Test Repo
2015/06/09 23:28:03 All indexes built!
2015/06/09 23:28:03 running server at http://localhost:6080...
2015/06/09 23:28:36 Rebuilding Test Repo for e33673116df7269fea9a2cd869fc70d4a75
72f65
2015/06/09 23:28:36 merge 0 files + mem
2015/06/09 23:28:36 155 data bytes, 1399 index bytes
HeapInUse = 0.79
HeapIdle  = 137.49
2015/06/09 23:29:40 Rebuilding Test Repo for 6092978b04fde85580a953a9736b4df325a
a9073
2015/06/09 23:29:40 merge 0 files + mem
2015/06/09 23:29:40 166 data bytes, 1431 index bytes
HeapInUse = 0.80
HeapIdle  = 137.45

**Comment by kellegous** [comment]
Awesome. This is working now. I also double checked that vmmap (and the windows equivalent) showed that we are properly unmapping files out of the memory space when we close an index.

**Pull request merged**